packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -105,12 +105,12 @@ export function createBufferedTransformStream(): TransformStream<
   const flushBuffer = (controller: TransformStreamDefaultController) => {
     if (!pendingFlush) {
       pendingFlush = new Promise((resolve) => {
-        setTimeout(() => {
+        queueTask(() => {
```

Comment:
It's not in the diff but queueTask is defined as:

```
const queueTask =
  process.env.NEXT_RUNTIME === 'edge' ? globalThis.setTimeout : setImmediate
```

at the top of this file